pkg/agent/util/net_linux.go

```diff
@@ -237,3 +241,172 @@ func DeleteOVSPort(brName, portName string) error {
 	cmd := exec.Command("ovs-vsctl", "--if-exists", "del-port", brName, portName)
 	return cmd.Run()
 }
+
+func GetNetRoutesOnAdapter(linkIndex int) (string, []interface{}, error) {
+	link, err := netlink.LinkByIndex(linkIndex)
+	if err != nil {
+		return "", nil, err
+	}
+	rs, err := netlink.RouteList(link, netlink.FAMILY_ALL)
+	if err != nil {
+		return "", nil, err
+	}
+	var gw string
+	var routes []interface{}
+	for _, r := range rs {
+		if r.Gw != nil && r.Dst != nil && r.Dst.IP.IsUnspecified() {
+			gw = r.Gw.String()
+		}
+		routes = append(routes, r)
+	}
+	return gw, routes, nil
+}
+
+func RenameInterface(from, to string) error {
+	var renameErr error
+	pollErr := wait.Poll(time.Millisecond*100, time.Second, func() (done bool, err error) {
```

Comment:
Add comment to explain that adding retry here because it might return `device is busy` error for one time configuration.